pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/block/question/PicklistCreatorHelper.java

```diff
@@ -0,0 +1,61 @@
+package org.broadinstitute.ddp.service.actvityinstancebuilder.block.question;
+
+import static org.broadinstitute.ddp.service.actvityinstancebuilder.util.TemplateHandler.addAndRenderTemplate;
+
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistGroupDef;
+import org.broadinstitute.ddp.model.activity.definition.question.PicklistOptionDef;
+import org.broadinstitute.ddp.model.activity.instance.question.PicklistGroup;
+import org.broadinstitute.ddp.model.activity.instance.question.PicklistOption;
+import org.broadinstitute.ddp.service.actvityinstancebuilder.AIBuilderContext;
+import org.broadinstitute.ddp.util.CollectionMiscUtil;
+
+
+public class PicklistCreatorHelper {
+
+    public PicklistGroup createPicklistGroup(AIBuilderContext ctx, PicklistGroupDef picklistGroupDef) {
+        return new PicklistGroup(
+                picklistGroupDef.getStableId(),
+                addAndRenderTemplate(ctx, picklistGroupDef.getNameTemplate())
+        );
+    }
+
+    public PicklistOption createPicklistOption(AIBuilderContext ctx, PicklistOptionDef picklistOptionDef,
+                                               List<PicklistGroupDef> picklistGroupDefs) {
+        return createPicklistOption(ctx, picklistOptionDef, picklistGroupDefs, false);
+    }
+
+    public PicklistOption createPicklistOption(AIBuilderContext ctx, PicklistOptionDef picklistOptionDef,
+                                               List<PicklistGroupDef> picklistGroupDefs, boolean isNested) {
+        PicklistOption picklistOption = new PicklistOption(
+                picklistOptionDef.getStableId(),
+                addAndRenderTemplate(ctx, picklistOptionDef.getOptionLabelTemplate()),
+                addAndRenderTemplate(ctx, picklistOptionDef.getTooltipTemplate()),
+                addAndRenderTemplate(ctx, picklistOptionDef.getDetailLabelTemplate()),
+                picklistOptionDef.isDetailsAllowed(),
+                picklistOptionDef.isExclusive(),
+                isNested ? null : addAndRenderTemplate(ctx, picklistOptionDef.getNestedOptionsLabelTemplate()),
+                isNested ? null :
+                    CollectionMiscUtil.createListFromAnotherList(picklistOptionDef.getNestedOptions(),
+                        (nestedPicklistOptionDef) -> createPicklistOption(ctx, nestedPicklistOptionDef,  picklistGroupDefs, true))
+        );
+
+        setGroupStableIdIfOptionBelongsToGroup(picklistGroupDefs, picklistOption);
+
+        return picklistOption;
+    }
+
+    private void setGroupStableIdIfOptionBelongsToGroup(List<PicklistGroupDef> picklistGroupDefs, PicklistOption picklistOption) {
```

Comment:
Might be better to have some sort of `Map<option_stable_id, group_def>` here instead of a `List` so we can turn this into `O(n)` (lookup in map is O(1) and we're doing N options) instead of `O(n^2)` (each lookup is O(n) and we're doing N options).